extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -443,6 +448,12 @@ const actions = (state, action, data) => {
         ...state,
         [StateProperty.VIEWER_CUSTOM_CONTROLS]: data,
       });
+    case Action.TOGGLE_SUBSCRIPTIONS_DIALOG:
+      return /** @type {!State} */ ({
+        ...state,
+        [StateProperty.SUBSCRIPTIONS_DIALOG_STATE]: !!data,
+        [StateProperty.PAUSED_STATE]: !!data,
```

Comment:
That could happen when the paywall is shown after 1 second on the paywall page, which it rarely happens given the short time of period the user has to pause right after navigating to the paywall page.

In addition, when we toggle false trying to hide the paywall and unpause the page, the PAUSED_STATE is always true and we cannot tell it is first paused by the user and then paused by this handler or it is 'only' paused by this handler. I think it's actually okay to unpause because I don't think the user would remember/notice/care it after running through the login/subscribe flow.

Reply:
Ok, let's just keep in mind the scenario where a user is already logged in and the result of the subscription takes a second to arrive. If we don't update the subscrption value at all, then it's ok to keep the logic as it is.

Would be good to keep this in mind and write very comprehensive tests so we cover all these scenarios (in future PRs).

Reply:
So the way `SubscriptionService` works is that when the extension loads it would fetch all entitlements. For AMP story case, when the time the user hits the paywall, the entitlements are already there so the delay is very unlikely.

Yeah agree. Next PR would be much more complex since it will involve `SubscriptionService` and we can discuss more about different scenarios there.

Reply:
Let's say you open the 4th page directly on a story, in that scenario the paywall would be hit immediately. SG to take care of this later though

Reply:
How can that happen? I thought you don't have a way to go to a specific page of the story because it is based on the historyState which is stored in local storage based on your visit history of the page?

Reply:
1) If you add `#page=<pageId>` to the URL, it will go directly to that page.
2 If you advance to page 3 and refresh, you start on that page.
3) If you view a story inside a player, the player can show any page at any time.